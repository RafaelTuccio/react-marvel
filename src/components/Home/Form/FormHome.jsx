import { useState } from "react";
import ic_busca from "../../assets/ic_busca.svg";



const FormHero = (props) => {
    let [name, setName] = useState("")
    let [error, setError] = useState("")
    let { heroHandle } = props

    return(
        <form className="form-group"
            onSubmit={(e) => {
            e.preventDefault();
            if(name){
                heroHandle({name});
            }else{
                setError("digite o nome de um heróis para começar a procurar")
            }
            
        }}>
            <label htmlFor=""><img src={ic_busca} alt="" /></label>
            <input type="text" 
            placeholder="Procure por um herói" 
            value={name}
            onChange={(e) => {
                setName(e.target.value);
                setError(null)
            }}
            />
            <span className="error">{error}</span>
        </form>
    );
}

export default FormHero;