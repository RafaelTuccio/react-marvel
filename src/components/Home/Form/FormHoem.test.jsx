import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import FormHome from './FormHome';

describe("<FormHome />", () => {
    it("deve renderizar normalmente na tela", () => {
        const {debug} = render(<FormHome />)
        debug()
    })

    it("deve chamar a função onChange toda ver que uma tecla for pressionada no input", () => {
        const fn = jest.fn();
        render(<FormHome heroHandle={fn}/>)
        const input = screen.getByPlaceholderText(/Procure por um herói/i)
        const value = "spider-man{enter}";

        userEvent.type(input, value)
        expect((fn)).toHaveBeenCalledTimes(1)
    })

    it("deve dar match com o snapshot", () => {
        const fn = jest.fn();
        const { container } = render(<FormHome heroHandle={fn}/>);
        expect(container.firstChild).toMatchSnapshot();
    })



    
})