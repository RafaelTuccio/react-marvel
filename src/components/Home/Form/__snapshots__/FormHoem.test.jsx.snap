// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormHome /> deve dar match com o snapshot 1`] = `
<form
  class="form-group"
>
  <label
    for=""
  >
    <img
      alt=""
      src="ic_busca.svg"
    />
  </label>
  <input
    placeholder="Procure por um herÃ³i"
    type="text"
    value=""
  />
  <span
    class="error"
  />
</form>
`;
